### PR TITLE
**Enhance property field handling with transformations**

### PR DESCRIPTION
**Pull request by gkrajniak**
- Added support for property field transformations in `FieldDefinition`, including `uppercase`, `lowercase`, `capitalize`, `encode`, and `decode`.
- Updated utilities like `getResourceValueByJsonPath` to apply these transformations.
- Enhanced documentation to reflect the new `propertyField` behavior.
- Added unit tests to validate transformations and edge cases.
- Updated dependencies in `package-lock.json` and `package.json`.

## Summary by CodeRabbit

* **New Features**
  * Added property-field display for nested object values with configurable text transforms (uppercase, lowercase, capitalize, encode, decode).

* **Documentation**
  * Expanded examples showing nested property selection and transform sequences in list views.

* **Tests**
  * Broadened test coverage for property extraction, transform sequences, Base64 encode/decode, and edge cases.

* **Chores**
  * Minor formatting and config adjustments; non-functional template formatting refinements.

### CODE REVIEW

File: docs/readme-generic-ui.md [hunk 1]
```diff
@@ -52,6 +52,14 @@ Each field definition supports the following properties:
 
 - `"label"`: Display name for the field
 - `"property"`: JSON path to the resource property (string or array of strings for fallback values)
+- `"propertyField"`: In case the property is a scalar value that represents an object, this property can be used to specify the field to be used for display within that object
```
Comment (gkrajniak) [inline docs/readme-generic-ui.md]: doc

**Review by nexus49 — APPROVED** [review]
